nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/gridfs/PutGridFS.java

```diff
@@ -212,9 +267,10 @@ private boolean canUploadFile(ProcessContext context, FlowFile input, String buc
 
             Document query;
             if (uniqueness.equals(UNIQUE_BOTH.getValue())) {
-                query = new Document().append("filename", fileName).append("md5", hash);
+                query = new Document().append("filename", fileName)
+                        .append("metadata", new Document(hashMetadataName, hash));
             } else if (uniqueness.equals(UNIQUE_HASH.getValue())) {
-                query = new Document().append("md5", hash);
+                query = new Document().append("metadata", new Document(hashMetadataName, hash));
```

Comment:
Is this change compatible with older versions of MongoDB?

Reply:
Yes and no. People who were relying on the `md5` attribute will have to come up with a new way of handling that on their existing data if they migrate it to our new approach. I think that can be mitigated by putting a notice in the migration guide. As you can see here, it does write the hash (which supports EL) as an attribute of the GridFS file metadata entry:

```
docker-rs:PRIMARY> db.put_test_bucket.files.find({}).pretty();
{
	"_id" : ObjectId("63532d0d56f7f5713dec57d6"),
	"filename" : "test_duplicates.txt",
	"length" : NumberLong(12),
	"chunkSize" : 262144,
	"uploadDate" : ISODate("2022-10-21T23:36:45.656Z"),
	"metadata" : {
		"hash" : "bc6e6f16b8a077ef5fbc8d59d0b931b9"
	}
}
```

Reply:
This might be worth a larger discussion on the dev mailing list as it probably makes more sense to support the feature as a custom deduplication query.

Reply:
Thanks for the reply @MikeThomsen. In order to avoid breaking existing users of PutGridFS, it seems like we either need to add server version checking, or perhaps a new configurable strategy property along the lines that you mentioned. Either way, we should have a path forward that maintains the current functionality, but also provides a new option that moves away from the deprecated `md5` approach.

Reply:
Given the potential complication, it seems better to focus this PR on the integration testing against current MongoDB versions, and then address the hashing change separately, which could include adjusting the integration tests to cover both scenarios.